### ISSUE TITLE
Adding Data from API limit 25

### ISSUE THREAD

**Issue by cizza** [gridsome/gridsome#400]
## Description

When I import data from an API  and add them to GraphQl data layer via api.loadSource method (as described in the docs) I can retrieve only 25 of them. 

### Steps to reproduce

Dunno how to reproduce, here's my code instead 

`module.exports = function (api) {
        api.loadSource(async store => {
            const {data} = await axios.get('http://localhost:3000/api/pages');

            const contentType = store.addContentType({
                typeName: 'MyPage'
            });

            for (const item of data) {
                contentType.addNode({
                    id: item.id,
                    title: item.name,
                    path: item.slug,
                    fields: {
                        name: item.name,
                        position: item.position,
                    }
                })
            }
        });
    }`

### Expected result
The expected result is to be able to retrieve all of MyPage in AllMyPage collection.

### Actual result

When I __explore the graphQl collection, I can only see 25 MyPage entries.

### Environment

<pre><code>
Libs:
- gridsome version: 5.8.0
- @gridsome/cli version:  n/a - gridsome installed via npm

Browser:
- [ ] Chrome (desktop) version latest
 
For Tooling issues:
- Node version: v11.12.0
- Platform:   Windows 
</code></pre>

If I specify contentType.addNode({ perPage: 100, ... then I'm able to retrieve all MyPage in allMyPage. However the real issue is when I try to build. I'm certain the build process is aware only of 25 pages and therefore I'm not able to proceed. 

When I query totalCount in the __explorer, it does show me 38 which is correct. However, when I dump the nodes there are only 25 and the MyPages which are not included do not show, when I run development. 

Any help or advice is more than welcome. Thank you.

**Comment by hjvedvik** [comment]
It's because of the `perPage` argument which defaults to `25` in the schema. I was just working on a fix for it by ignoring `perPage` if no `page` argument is provided. As a workaround now, you can set `perPage` to something else:

```graphql
query {
  allMyPage(perPage: 999) {
    edges {
      node {
        id
      }
    }
  }
}
```

**Comment by cizza** [comment]
Hello Hans-Jørgen, 

thank you for your prompt reply. I'm afraid I don't think that helps me. I don't query allMyPage anywhere in my code. I fetch the data from API, build up the MyPage collection and that's what I query on my pages.

`<page-query>
query page {
    myPage(path:"/home") {
        name
        title
        slug
        position ... 
  }
}
</page-query>`

I do not use the allMyPage anywhere in my code, so I can't set the "perPage" ad hoc. 
It maybe sounds like that I don't need to but I believe this is the reason I can't build my site, as it looks like to me that the build process doesn't have certain myPage in the collection.  Does it make sense?

Thank you.

**Comment by cizza** [comment]
I found the PER_PAGE constant and changed it manually. Not healthy but ok for troubleshooting. At least I'm able to tell that the reason why gridsome doesn't build the pages is not here. I'm experiencing the closed issue #366 - `gridsome build` hangs after Compile assets;

**Comment by cizza** [comment]
I found my bug, its actually unrelated to the described issue. Thank you
